Android API30+ Triage due to Loss of NetlinkSockets (#2237)
objectcomputing/OpenDDS: docs/android.md

```Shell
./configure --doc-group --target=android --macros=ANDROID_ABI=armeabi-v7a

>     --macros=android_sdk=$SDK --macros=android_target_api=$API

Reviewer: This is just a minimal example and doesn't do anything because `--java` isn't being passed. If you want to keep it I would suggest moving this to the "Network Permissions and Availability" section.

(cellular or wifi) to another. Apps that need to know when a network change
occurs can register with the [ConnectivityManager](https://developer.android.com/reference/android/net/ConnectivityManager) for network callback events.

As of API 30+, LinuxNetworkConfigMonitor can no longer be used, as Netlink
sockets are blocked by OS for security reasons. Instead, NetworkConfigModifier
is utilized.  As a consequence of this, two new variables are required from the
user, android_sdk, and android_target_api.  These correspond to the location of your
sdk, likely /home/<username>/Android/Sdk on Linux, and the API number you are targeting.
The NetworkConfigModifier is set up along with the necessary network callbacks
when the user uses `TheParticipantFactory.WithArgs`.

```Shell
./configure --doc-group --target=android --macros=ANDROID_ABI=armeabi-v7a

>     --macros=android_sdk=$SDK --macros=android_target_api=$API --java

Reviewer: So the target API passed here will almost certainly be different from the minimum API that is passed to `make_standalone_toolchain.py`. I would suggest using a different variable name.

Reviewer: `$API` that is.

(cellular or wifi) to another. Apps that need to know when a network change
occurs can register with the [ConnectivityManager](https://developer.android.com/reference/android/net/ConnectivityManager) for network callback events.

As of API 30+, LinuxNetworkConfigMonitor can no longer be used, as Netlink 
sockets are blocked by OS for security reasons. Instead, NetworkConfigModifier

> is utilized.  As a concequence of this, two new variables are required from the 

Reviewer: ```suggestion
is utilized.  As a consequence of this, two new variables are required from the 
```

user, android_sdk, and android_target_api.  These correspond to the location of your

> sdk, likely /home/<username>/Android/Sdk on Linux, and the API number you are building
> for. The NetworkConfigModifier is set up along with the necessary network callbacks

Reviewer: ```suggestion
sdk, likely /home/<username>/Android/Sdk on Linux, and the API number you are targeting.
The NetworkConfigModifier is set up along with the necessary network callbacks
```

> when the user uses "TheParticipantFactory.WithArgs".

Reviewer: ```suggestion
when the user uses `TheParticipantFactory.WithArgs`.
```
